src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -2236,7 +2239,10 @@ impl ExecutorContext {
                 )
             })?;
         // Move the artifact commands to simplify cache management.
-        exec_state.global.artifact_commands = self.engine.take_artifact_commands();
+        exec_state
+            .global
+            .artifact_commands
+            .extend(self.engine.take_artifact_commands());
```

Comment:
This is the key part yeah? Extending the old `artifact_commands` with the new ones?

Reply:
Yes. This is the one line I needed.